src/components/select/option/option.component.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import StyledOption from './option.style';
+
+const Option = ({
+  text,
+  children,
+  onSelect,
+  value,
+  index,
+  selectedIndex
+}) => {
+  const [isHighlighted, setIsHighlighted] = useState();
+  useEffect(() => {
+    setIsHighlighted(index === selectedIndex);
+  }, [index, selectedIndex]);
+
+  function handleClick() {
+    onSelect({ text, value });
+  }
+
+  return (
+    <StyledOption
+      aria-selected={ isHighlighted }
+      data-component='option'
+      onClick={ handleClick }
+      isHighlighted={ isHighlighted }
+      role='option'
+    >
+      { children || text }
+    </StyledOption>
+  );
+};
+
+Option.propTypes = {
+  /** The option's visible text, displayed within Textbox of Select, and used for filtering */
+  text: PropTypes.string.isRequired,
+  /** Optional: alternative rendered content, displayed within SelectList of Select (eg: an icon, an image, etc) */
+  children: PropTypes.node,
+  /** The option's invisible internal value */
+  value: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
+  /** Callback to return value when the element is selected (prop added by the SelectList component) */
+  onSelect: PropTypes.func,
+  /** Index of the option in the SelectList (prop added by the SelectList component) */
+  index: PropTypes.number,
+  /** Currently selected index (prop added by the SelectList component) */
+  selectedIndex: PropTypes.number
```

Comment:
We should use the `@ignore` and `@private` jsdoc annotation for the internal prop types. This has the added benefit that they won't be included in the prop tables, no need to ignore them in MDX.

Reply:
We need `@ignore`.

Reply:
added `@ignore`